tests/simulators/metadrive/test_metadrive.py

```diff
@@ -0,0 +1,131 @@
+    scenario = tryPickling(loadLocalScenario("basic.scenic", mode2D=True))
+    tryPickling(sampleScene(scenario, maxIterations=1000))
+
+
+@pytest.fixture(scope="package")
+def getMetadriveSimulator(getAssetPath):
+    base = getAssetPath("maps/CARLA")
+
+    def _getMetadriveSimulator(town):
+        openDrivePath = os.path.join(base, f"{town}.xodr")
+        sumoPath = os.path.join(base, f"{town}.net.xml")
+        simulator = MetaDriveSimulator(sumo_map=sumoPath, render=False)
+        return simulator, openDrivePath, sumoPath
+
+    yield _getMetadriveSimulator
+
+
+def test_throttle(getMetadriveSimulator):
+    simulator, openDrivePath, sumoPath = getMetadriveSimulator("Town01")
+    code = f"""
+        param map = r'{openDrivePath}'
+        param sumo_map = r'{sumoPath}'
+
+        model scenic.simulators.metadrive.model
+
+        behavior DriveWithThrottle():
+            while True:
+                take SetThrottleAction(1)
+
+        ego = new Car at (369, -326), with behavior DriveWithThrottle
+        record ego.speed as CarSpeed
+        terminate after 5 steps
+    """
+    scenario = compileScenic(code, mode2D=True)
+    scene = sampleScene(scenario)
+    simulation = simulator.simulate(scene)
+    records = simulation.result.records["CarSpeed"]
+    assert records[len(records) // 2][1] < records[-1][1]
+
+
+@pytest.mark.skip(
+    reason="This test is skipped until MetaDrive uploads the next version on PyPI to fix the issue where cars aren't fully stopping."
+)
+def test_brake(getMetadriveSimulator):
+    simulator, openDrivePath, sumoPath = getMetadriveSimulator("Town01")
+    code = f"""
+        param map = r'{openDrivePath}'
+        param sumo_map = r'{sumoPath}'
+
+        model scenic.simulators.metadrive.model
+
+        behavior DriveWithThrottle():
+            while True:
+                take SetThrottleAction(1)
+
+        behavior Brake():
+            while True:
+                take SetThrottleAction(0), SetBrakeAction(1)
+
+        behavior DriveThenBrake():
+            do DriveWithThrottle() for 2 steps
+            do Brake() for 6 steps
+
+        ego = new Car at (369, -326),
+            with behavior DriveThenBrake
+        record final ego.speed as CarSpeed
+        terminate after 8 steps
+    """
+    scenario = compileScenic(code, mode2D=True)
+    scene = sampleScene(scenario)
+    simulation = simulator.simulate(scene)
+    finalSpeed = simulation.result.records["CarSpeed"]
+    assert finalSpeed == pytest.approx(0.0, abs=1e-1)
+
+
+def test_pedestrian_movement(getMetadriveSimulator):
+    simulator, openDrivePath, sumoPath = getMetadriveSimulator("Town01")
+    code = f"""
+        param map = r'{openDrivePath}'
+        param sumo_map = r'{sumoPath}'
+
+        model scenic.simulators.metadrive.model
+
+        behavior WalkForward():
+            while True:
+                take SetWalkingDirectionAction(self.heading)
+                take SetWalkingSpeedAction(0.5)
```

Comment:
```suggestion
                take SetWalkingDirectionAction(self.heading), SetWalkingSpeedAction(0.5)
```
This may break, if the actions interfere with each other as we discussed, but it's supposed to be possible to use these actions simultaneously (see the `WalkForwardBehavior` in CARLA, for example).

Reply:
Updated in commit 89878f32b7a4dcb66c54d287524193c0d030c1be